tests/unit/events/test_event_manager.py

```diff
@@ -83,11 +89,36 @@ async def test_emit_event_with_no_listeners(
 
     # Attempt to emit an event for which no listeners are registered, it should not fail
     event_manager.emit(event=Event.SYSTEM_INFO, event_data=event_system_info_data)
+    await asyncio.sleep(0.1)  # Allow some time for the event to be processed
 
     # Ensure the listener for the other event was not called
     assert async_listener.call_count == 0
 
 
+async def test_emit_invokes_parameterless_listener(
+    event_manager: EventManager,
+    event_system_info_data: EventSystemInfoData,
+) -> None:
+    sync_mock = MagicMock()
+
+    def sync_listener() -> None:
+        sync_mock()
+
+    async_mock = MagicMock()
+
+    async def async_listener() -> None:
+        async_mock()
+
+    event_manager.on(event=Event.SYSTEM_INFO, listener=sync_listener)
+    event_manager.on(event=Event.SYSTEM_INFO, listener=async_listener)
+
+    event_manager.emit(event=Event.SYSTEM_INFO, event_data=event_system_info_data)
+    await asyncio.sleep(0.1)  # Allow some time for the event to be processed
```

Comment:
This probably will not be a problem in this case. But I would suggest to adopt a specific pattern in places where we use some time sleep(which we can´t rely on in our context) and have some expectation that should become true after some sleep time. 

Advantages of such pattern is that minimal time sleep can be very low. For example 0.01, but it can define large tolerance, like sleep 1s to make the test more robust.

I used it recently when I saw test that is locally stable to fail in CI. It could be abstracted and used in many places in our tests to avoid random CI failures and speed up tests on average.
https://github.com/apify/crawlee-python/pull/810/files

(Not strictly related to this change.)

Reply:
Could you make a separate issue for that please?

Reply:
https://github.com/apify/crawlee-python/issues/824